File: src/main/java/management/resident_management/entity/Fee.java
package management.resident_management.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "fees")
public class Fee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "apartment_id", nullable = false)
    private Apartment apartment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "fee_type_id", nullable = true)
    private FeeType feeType;

    @Column
    private Long residentId;

    @Column(nullable = false)
    private Double amount;

    @Column(nullable = false)
    private LocalDate dueDate;

    @Column
    private LocalDate paymentDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FeeStatus status = FeeStatus.UNPAID;

    @Column
    private String description;

    @Column
    private String stripePaymentIntentId;

    @Column
    private String stripePaymentStatus;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = FeeStatus.UNPAID;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: billing-frontend/src/types/fee.ts
export interface FeeDTO {
    id: number;
    apartmentId: number;
    feeTypeId: number;
    residentId: number;
    amount: number;
    dueDate: string;
    paymentDate: string | null;
    status: 'PAID' | 'UNPAID' | 'OVERDUE';
    description: string | null;
    stripePaymentIntentId: string | null;
    stripePaymentStatus: string | null;
    createdAt: string;
    updatedAt: string;
}

export interface NewFeeDTO {
    apartmentId: number;
    feeTypeId: number;
    amount: number;
    dueDate: string;
    description: string | null;
    status: 'PAID' | 'UNPAID' | 'OVERDUE';
    residentId: number;
}

export interface PaymentRequestDTO {
    feeId: string;
    residentId: string;
    amount: number;
    paymentMethodId: string;
}

export interface RefundRequestDTO {
    feeId: string;
    paymentIntentId: string;
}